### ISSUE TITLE
Edit article page does not keep assigned template

### ISSUE THREAD

**Issue by osignell** [joomla/joomla-cms#16598]
### Steps to reproduce the issue
Create an article A.
Create menu item 1 of the type "single article" that points to the article A.
Assign a separate template to menu item 1.
Create menu item 2 of the type "create article"
Open page 1.
Click "edit" on the article.

### Expected result
Article editor for article A opened with the template and modules assigned to menu item 1.

### Actual result
Article editor for article A opened with Menu item 2 and the layout and modules for that menu item.

### System information (as much as possible)

Joomla 3.7.2

### Additional comments

This is probably related to #12687 ?

This used to work in Joomla 3.6, ie the menu item id for the page with the edit button was used when showing the editor.

**Comment by brianteeman** [comment]
That is the expected behaviour as you have defined a template for the edit view and that is being loaded when you try to edit the article

**Comment by franz-wohlkoenig** [comment]
Closed as expected Behaviour.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16598">issues.joomla.org/tracker/joomla-cms/16598</a>.</sub>

**Comment by osignell** [comment]
Let me elaborate a bit... Let's say I have a site with two article categories and two blog view pages, one for each category. Each blog view has its own template.

I would like to make it easy for the persons creating content, so I created "add article" pages below each of the blog views and set these up with a "default category" to force the articles into the right categories. I also selected the same templates as the blog views to make it consistent.

This is all well and good, except..
when an existing article (any article) is edited, the itemid used is always the first create article menu item (with the lower id) regardless of any "default category" settings. Example:
- I open the category 2 blog view - I get template2
- I create an article there with the dedicated create men item - I get template2
- I open a single article in its entirety - I get template2
- I try to edit this article - I suddenly get template 1?
